src/ConstructiveSolidGeometry/VolumePrimitives/Box.jl

```diff
@@ -1,84 +1,78 @@
-struct Box{T,TX,TY,TZ} <: AbstractVolumePrimitive{T}
-    x::TX
-    y::TY
-    z::TZ
-    
-    function Box( ::Type{T},
-                  x::Union{T, <:AbstractInterval{T}},
-                  y::Union{T, <:AbstractInterval{T}},
-                  z::Union{T, <:AbstractInterval{T}}) where {T}
-        new{T,typeof(x),typeof(y),typeof(z)}(x, y, z)
-    end
-end
+"""
+    Box{T, CO} <: AbstractVolumePrimitive{T}
 
-#Constructors
-function Box(;xMin = -1, xMax = 1, yMin = -1, yMax = 1, zMin = -1, zMax = 1)
-    T = float(promote_type(typeof.((xMin, xMax, yMin, yMax, zMin, zMax))...))
-    x = xMax == -xMin ? T(xMax) : T(xMin)..T(xMax)
-    y = yMax == -yMin ? T(yMax) : T(yMin)..T(yMax)
-    z = zMax == -zMin ? T(zMax) : T(zMin)..T(zMax)
-    Box( T, x, y, z)
+T: Type of values, e.g. Float64
+CO: ClosedPrimitive or OpenPrimitive <-> whether surface belongs to it or not
+"""
+@with_kw struct Box{T, CO} <: AbstractVolumePrimitive{T, CO}
+    hX::T = 1
+    hY::T = 1
+    hZ::T = 1
+    origin::CartesianPoint{T} = zero(CartesianPoint{T})
+    rotation::SMatrix{3,3,T,9} = one(SMatrix{3, 3, T, 9})
 end
-Box(xMin, xMax, yMin, yMax, zMin, zMax) = Box(; xMin = xMin, xMax = xMax, yMin = yMin, yMax = yMax, zMin = zMin, zMax = zMax)
 
-function Box(x::X, y::Y, z::Z) where {X<:Real, Y<:Real, Z<:Real}
-    T = float(promote_type(X,Y,Z))
-    Box( T, T(x/2), T(y/2), T(z/2))
+Box{T, CO}( b::Box{T, CO}; COT = CO,
```

Comment:
Is this intended that the keyword's name is `COT`?
I would naively assume that `Box{T, COT}( b::Box{T, COT}; CO = COT, ..} where {COT <: ...}` would be more intuitive.